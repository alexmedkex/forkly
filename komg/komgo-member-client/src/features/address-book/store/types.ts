import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
