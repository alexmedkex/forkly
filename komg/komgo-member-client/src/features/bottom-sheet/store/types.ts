import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
