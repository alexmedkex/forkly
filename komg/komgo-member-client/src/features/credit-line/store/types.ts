import { ImmutableMap } from '../../../utils/types'
import { Map } from 'immutable'
import {
  ICreditLine,
  IRiskCoverData,
  Currency,
  ICreditLineResponse,
  ISharedCreditLine,
  IProductContext,
  ICreditLineRequest,
  ICreateCreditLineRequest,
  IInformationShared
} from '@komgo/types'
import { Products } from '../../document-management/constants/Products'
import { SubProducts } from '../../document-management/constants/SubProducts'
import { IMember } from '../../members/store/types'

export enum CreditLineActionType {
  FetchCreditLinesRequest = '@@risk-cover/FETCH_CREDIT_LINES_REQUEST',
  FetchCreditLinesSuccess = '@@risk-cover/FETCH_CREDIT_LINES_SUCCESS',
  FetchCreditLinesFailure = '@@risk-cover/FETCH_CREDIT_LINES_FAILURE',
  GetCreditLineRequest = '@@risk-cover/GET_CREDIT_LINE_REQUEST',
  GetCreditLineSuccess = '@@risk-cover/GET_CREDIT_LINE_SUCCESS',
  GetCreditLineFailure = '@@risk-cover/GET_CREDIT_LINE_FAILURE',
  CreateCreditLineRequest = '@@risk-cover/CREATE_CREDIT_LINE_REQUEST',
  CreateCreditLineSuccess = '@@risk-cover/CREATE_CREDIT_LINE_SUCCESS',
  CreateCreditLineFailure = '@@risk-cover/CREATE_CREDIT_LINE_FAILURE',
  EditCreditLineRequest = '@@risk-cover/EDIT_CREDIT_LINE_REQUEST',
  EditCreditLineSuccess = '@@risk-cover/EDIT_CREDIT_LINE_SUCCESS',
  EditCreditLineFailure = '@@risk-cover/EDIT_CREDIT_LINE_FAILURE',
  FetchDisclosedCreditLineSummariesRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_REQUEST',
  FetchDisclosedCreditLineSummariesSuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_SUCCESS',
  FetchDisclosedCreditLineSummariesFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_FAILURE',
  FetchDisclosedCreditLinesForCounterpartyRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_REQUEST',
  FetchDisclosedCreditLinesForCounterpartySuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_SUCCESS',
  FetchDisclosedCreditLinesForCounterpartyFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_FAILURE',
  RemoveCreditLineRequest = '@@risk-cover/REMOVE_CREDIT_LINE_REQUEST',
  RemoveCreditLineSuccess = '@@risk-cover/REMOVE_CREDIT_LINE_SUCCESS',
  RemoveCreditLineFailure = '@@risk-cover/REMOVE_CREDIT_LINE_FAILURE',
  CreateReqInformationRequest = '@@risk-cover/CREATE_REQ_INFORMATION_REQUEST',
  CreateReqInformationSuccess = '@@risk-cover/CREATE_REQ_INFORMATION_SUCCESS',
  CreateReqInformationFailure = '@@risk-cover/CREATE_REQ_INFORMATION_FAILURE',
  FetchRequestsRequest = '@@risk-cover/FETCH_REQ_REQUEST',
  FetchRequestsSuccess = '@@risk-cover/FETCH_REQ_SUCCESS',
  FetchRequestsFailure = '@@risk-cover/FETCH_REQ_FAILURE',
  DeclineAllRequestsRequest = '@@risk-cover/DECLINE_ALL_REQ_REQUEST',
  DeclineAllRequestsSuccess = '@@risk-cover/DECLINE_ALL_REQ_SUCCESS',
  DeclineAllRequestsFailure = '@@risk-cover/DECLINE_ALL_REQ_FAILURE'
}

export enum CreditLineType {
  RiskCover = 'riskCover',
  BankLine = 'bankLine'
}

export interface IExtendedCreditLine extends ICreditLineResponse {
  counterpartyName: string
  counterpartyLocation: string
  data: IRiskCoverData
  sharedCreditLines: IExtendedSharedCreditLine[]
}

export interface IExtendedSharedCreditLine extends ISharedCreditLine<IInformationShared> {
  updatedAt?: string
  counterpartyName: string
}

export interface ICreateOrEditSharedCreditLine {
  counterpartyStaticId: string
  sharedWithStaticId: string
  data: IInformationShared
  requestStaticId?: string
  staticId?: string
  updatedAt?: string
}
export interface ICreateOrEditCreditLineForm {
  context: IProductContext
  counterpartyStaticId: string
  appetite: boolean
  availability: boolean
  creditLimit?: number
  availabilityAmount?: number
  data?: IRiskCoverData
  currency: Currency
  sharedCreditLines: ICreateOrEditSharedCreditLine[]
  availabilityAmountUpdatedAt?: string | Date
  staticId?: string
  creditExpiryDate: string | Date
}

export interface IProductProps {
  productId: Products
  subProductId: SubProducts
}

// TODO: move to komgo types
export interface IDisclosedCreditLineSummary {
  counterpartyStaticId: string
  lowestFee: number
  availabilityCount: number
  appetiteCount: number
  _id: string
}

export interface IExtendedCreditLineRequest extends ICreditLineRequest {
  counterpartyName: string
  companyName: string
}

export interface IDisclosedCreditLineSummaryEnriched extends IDisclosedCreditLineSummary {
  counterpartyName: string
  counterpartyLocation: string
}

export interface IDisclosedCreditLineEnriched extends IDisclosedCreditLine {
  companyName: string
  companyLocation: string
  counterpartyName: string
}

export interface IDisclosedCreditLine extends ICreditLine {
  ownerStaticId: string
  data?: any // TODO: move later to the right type inside `ICreditLine`
}

export interface IRequestInformationForm {
  mailTo: boolean
  comment: string
  requestForId: string
  companyIds: string[]
}

export interface IRequestCreditLineForm extends IRequestInformationForm {
  context: IProductContext
}

export interface IMailToData {
  email?: string
  subject?: string
  body?: string
}

export interface RiskCoverProperties {
  creditLinesById: Map<string, ICreditLineResponse>
  disclosedCreditLineSummariesById: Map<string, IDisclosedCreditLineSummary>
  disclosedCreditLinesById: Map<string, IDisclosedCreditLine>
  requestsById: Map<string, ICreditLineRequest>
}

export interface CreditLinesProperties {
  [CreditLineType.RiskCover]: ImmutableMap<RiskCoverProperties>
  [CreditLineType.BankLine]: ImmutableMap<RiskCoverProperties>
}

export type CreditLinesState = ImmutableMap<CreditLinesProperties>

export interface IMemberWithDisabledFlag extends IMember {
  disabled?: boolean
}
