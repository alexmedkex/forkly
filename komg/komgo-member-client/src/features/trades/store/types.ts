import { IPaginate, ImmutableMap } from '../../../utils/types'
import { Map, List } from 'immutable'

import { stringOrNull } from '../../../utils/types'
import { ITradeBase, ICargoBase, ITrade, ICargo } from '@komgo/types'

import { Action } from 'redux'
import { TradingRole } from '../../trades/constants'
import { Document } from '../../document-management'

export enum TradeActionType {
  TRADE_REQUEST = '@@trades/TRADE_REQUEST',
  TRADE_SUCCESS = '@@trades/TRADE_SUCCESS',
  TRADE_FAILURE = '@@trades/TRADE_FAILURE',
  TRADE_MOVEMENTS_REQUEST = '@@trades/TRADE_MOVEMENTS_REQUEST',
  TRADE_MOVEMENTS_SUCCESS = '@@trades/TRADE_MOVEMENTS_SUCCESS',
  TRADE_MOVEMENTS_FAILURE = '@@trades/TRADE_MOVEMENTS_FAILURE',
  TRADE_DOCUMENTS_REQUEST = '@@trades/TRADE_DOCUMENTS_REQUEST',
  TRADE_DOCUMENTS_SUCCESS = '@@trades/TRADE_DOCUMENTS_SUCCESS',
  TRADE_DOCUMENTS_FAILURE = '@@trades/TRADE_DOCUMENTS_FAILURE',
  TRADES_REQUEST = '@@trades/TRADES_REQUEST',
  TRADES_SUCCESS = '@@trades/TRADES_SUCCESS',
  TRADES_FAILURE = '@@trades/TRADES_FAILURE',
  TRADE_TOTAL_FETCHED = '@@trades/TRADE_TOTAL_FETCHED',
  SORT_TRADES = '@@trades/SORT_TRADES',
  FILTER_TRADING_ROLE = '@@trades/FILTER_TRADING_ROLE',
  CREATE_TRADE_REQUEST = '@@trades/CREATE_TRADE_REQUEST',
  CREATE_TRADE_FAILURE = '@@trades/CREATE_TRADE_FAILURE',
  CREATE_TRADE_SUCCESS = '@@trades/CREATE_TRADE_SUCCESS',
  UPLOAD_TRADE_DOCUMENT_REQUEST = '@@trades/UPLOAD_TRADE_DOCUMENT_REQUEST',
  UPLOAD_TRADE_DOCUMENT_SUCCESS = '@@trades/UPLOAD_TRADE_DOCUMENT_SUCCESS',
  UPLOAD_TRADE_DOCUMENT_FAILURE = '@@trades/UPLOAD_TRADE_DOCUMENT_FAILURE',
  EDIT_TRADE_REQUEST = '@@trades/EDIT_TRADE_REQUEST',
  EDIT_TRADE_FAILURE = '@@trades/EDIT_TRADE_FAILURE',
  EDIT_TRADE_SUCCESS = '@@trades/EDIT_TRADE_SUCCESS',
  DELETE_TRADE_REQUEST = '@@trades/DELETE_TRADE_REQUEST',
  DELETE_TRADE_FAILURE = '@@trades/DELETE_TRADE_FAILURE',
  DELETE_TRADE_SUCCESS = '@@trades/DELETE_TRADE_SUCCESS',
  CREATE_CARGO_REQUEST = '@@trades/CREATE_CARGO_REQUEST',
  CREATE_CARGO_FAILURE = '@@trades/CREATE_CARGO_FAILURE',
  CREATE_CARGO_SUCCESS = '@@trades/CREATE_CARGO_SUCCESS',
  EDIT_CARGO_REQUEST = '@@trades/EDIT_CARGO_REQUEST',
  EDIT_CARGO_FAILURE = '@@trades/EDIT_CARGO_FAILURE',
  EDIT_CARGO_SUCCESS = '@@trades/EDIT_CARGO_SUCCESS',
  DELETE_CARGO_REQUEST = '@@trades/DELETE_CARGO_REQUEST',
  DELETE_CARGO_FAILURE = '@@trades/DELETE_CARGO_FAILURE',
  DELETE_CARGO_SUCCESS = '@@trades/DELETE_CARGO_SUCCESS',
  DELETE_TRADE_DOCUMENT_REQUEST = '@@trades/DELETE_TRADE_DOCUMENT_REQUEST',
  DELETE_TRADE_DOCUMENT_SUCCESS = '@@trades/DELETE_TRADE_DOCUMENT_SUCCESS',
  DELETE_TRADE_DOCUMENT_FAILURE = '@@trades/DELETE_TRADE_DOCUMENT_FAILURE'
}

export interface CreateTradeDocumentRequest {
  context: TradeDocumentContext
  name: string
  categoryId: string
  documentTypeId: string
  file: File
  owner: {
    firstName: string
    lastName: string
    companyId: string
  }
}

export interface TradeDocumentContext extends DocumentContextBase {
  vaktId: string
}

export interface DocumentContextBase {
  productId: string
  subProductId: string
}

export interface TradeStateProperties {
  trades: Map<string, ITrade>
  tradeDocuments: List<Document>
  tradeMovements: List<ICargo>
  tradeIds: List<string>
  totals: Map<string, number>
  error: stringOrNull
  confirmError: stringOrNull
}

export interface TableSortParams {
  column: keyof ITradeEnriched
  direction: number
}

export interface TableFilterParams {
  role: TradingRole
  company: string
}

export interface IPaymentsTerms {
  eventBase: string
  when: string
  time: number
  timeUnit: string
  dayType: string
}

export interface IPeriod {
  startDate: string | number | Date
  endDate: string | number | Date
}
export interface ITradeEnriched extends ITrade {
  buyerName?: string
  sellerName?: string
  tradingRole?: TradingRole
}

export interface ITradeDocument {
  _id?: string
  id?: string
  name: string
  categoryId?: string
  typeId: string
  file: File
  fileName: string
  fileType: string
}

export interface ICreateOrUpdateTrade {
  trade: ITradeEnriched
  cargo: ICargo
  documents: ITradeDocument[]
  lawOther?: string
  commodityOther?: string
  eventBaseOther?: string
  deliveryTermsOther?: string
}

export interface IActor {
  id: string
  key: string
}

export type TradeState = ImmutableMap<TradeStateProperties>

export interface TradesReceivedAction extends Action {
  meta: any
  type: TradeActionType.TRADES_SUCCESS
  payload: IPaginate<ITrade>
}

export interface TradeReceivedAction extends Action {
  type: TradeActionType.TRADE_SUCCESS
  payload: ITrade
}

export interface TradeTotalsAction extends Action {
  meta: any
  type: TradeActionType.TRADE_TOTAL_FETCHED
  payload: IPaginate<ITrade>
}

export interface TradesMovementsReceivedAction extends Action {
  type: TradeActionType.TRADE_MOVEMENTS_SUCCESS
  payload: ICargo[]
}

export interface TradesError extends Action {
  type: TradeActionType.TRADES_FAILURE | TradeActionType.TRADE_FAILURE | TradeActionType.TRADE_MOVEMENTS_FAILURE
  payload: stringOrNull
}

export interface SortTrades extends Action {
  type: TradeActionType.SORT_TRADES
  payload: TableSortParams
}

export interface FilterTradingRole extends Action {
  type: TradeActionType.FILTER_TRADING_ROLE
  payload: TableFilterParams
}

export interface CreateTradeError extends Action {
  type: TradeActionType.CREATE_TRADE_FAILURE
  payload: stringOrNull
}

export interface CreateCargoError extends Action {
  type: TradeActionType.CREATE_CARGO_FAILURE
  payload: stringOrNull
}

export interface TradeSuccessfullyCreated extends Action {
  type: TradeActionType.CREATE_TRADE_SUCCESS
  payload: string
}

export interface DeleteTradeError extends Action {
  type: TradeActionType.DELETE_TRADE_FAILURE
  payload: string
}

export interface EditTradeError extends Action {
  type: TradeActionType.EDIT_TRADE_FAILURE
  payload: string
}

export interface EditCargoError extends Action {
  type: TradeActionType.EDIT_CARGO_FAILURE
  payload: string
}

export type TradeAction =
  | TradesReceivedAction
  | TradesError
  | SortTrades
  | FilterTradingRole
  | TradeReceivedAction
  | TradesMovementsReceivedAction
  | TradeTotalsAction
  | CreateTradeError
  | TradeSuccessfullyCreated
  | DeleteTradeError
  | EditTradeError
