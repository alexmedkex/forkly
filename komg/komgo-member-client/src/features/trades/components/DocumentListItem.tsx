import * as React from 'react'
import styled from 'styled-components'
import { DocumentResponse, HasId } from '../../document-management'
import { EllipsisDropdown } from '../../document-management/components/dropdowns/EllipsisDropdown'
import { DropdownOption } from '../../document-management/components/documents/my-documents/DocumentListDropdownOptions'
import { Image, List } from 'semantic-ui-react'
import Text from '../../../components/text'

export interface IDocumentItem extends HasId {
  name: string
  typeName: string
}

interface DocumentListItemProps {
  document: IDocumentItem
  renderDropdownOptions: (document: IDocumentItem) => DropdownOption[]
}

export const DocumentListItem: React.FC<DocumentListItemProps> = ({ document, renderDropdownOptions }) => (
  <List.Item data-test-id={document.name}>
    <List.Content floated="right">
      <EllipsisDropdown options={renderDropdownOptions} item={document} style={{ paddingRight: 0 }} />
    </List.Content>
    <Image src="/images/file.svg" inline={true} spaced="right" />
    <List.Content>
      <Text bold={true}>{document.name}</Text>
    </List.Content>
    <StyledContentMiddle>{document.typeName}</StyledContentMiddle>
  </List.Item>
)

const StyledContentMiddle = styled(List.Content)`
  display: inline;
  left: 50%;
  margin-right: -50%;
  position: absolute;
  text-align: left;
  line-height: 1.9rem !important;
`
