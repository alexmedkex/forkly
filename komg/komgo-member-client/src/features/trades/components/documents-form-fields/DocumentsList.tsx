import * as React from 'react'
import { SyntheticEvent } from 'react'
import { List } from 'semantic-ui-react'
import { ITradeDocument } from '../../store/types'
import { DocumentType } from '../../../document-management/store/types/document-type'
import { IDocumentItem, DocumentListItem } from '../DocumentListItem'
import styled from 'styled-components'

import {
  DropdownOption,
  DELETE,
  EDIT
} from '../../../document-management/components/documents/my-documents/DocumentListDropdownOptions'

const Padding = styled.div`
  padding-top: 4px;
`

export interface IProps {
  documents: ITradeDocument[]
  documentTypes: DocumentType[]
  inForm?: boolean
  removeDocument?: (index: number) => void
  editDocument?: (index: number) => void
}

interface IState {
  actives: any
}

export default class DocumentsList extends React.Component<IProps, IState> {
  static formatDocumentTypes = (documentTypes: DocumentType[]) => {
    return documentTypes.reduce((memo, documentType) => {
      return {
        ...memo,
        [documentType.id]: documentType
      }
    }, {})
  }

  constructor(props: IProps) {
    super(props)
    this.state = {
      actives: this.props.documents.reduce((memo, document) => {
        return {
          ...memo,
          [document._id!]: true
        }
      }, {})
    }
  }

  handleClick = (e: SyntheticEvent, titleProps: any) => {
    const { index } = titleProps
    const { actives } = this.state
    this.setState({
      actives: {
        ...actives,
        [index]: !actives[index]
      }
    })
  }

  renderDropdownOptions = (index: number, document: ITradeDocument): DropdownOption[] => {
    const { removeDocument, editDocument } = this.props

    const view: DropdownOption = {
      ...EDIT,
      onClick: () => {
        editDocument(index)
      }
    }

    const download: DropdownOption = {
      ...DELETE,
      onClick: () => {
        removeDocument(index)
      }
    }

    const dropdownOptions = document.file ? { download, view } : { download }
    return Object.values(dropdownOptions)
  }

  createDocumentItem = (document: ITradeDocument): IDocumentItem => {
    const typeName: string = DocumentsList.formatDocumentTypes(this.props.documentTypes)[document.typeId].name
    const fileNameSplit: string[] = document.fileName.split('.')
    const fileExtension: string = fileNameSplit.length > 1 ? '.' + fileNameSplit[1] : ''
    const fileName: string = document.name + fileExtension
    return {
      id: document.id,
      name: fileName,
      typeName
    }
  }

  render() {
    const hasDocuments = this.props.documents && this.props.documents.length > 0
    return (
      <div>
        <List divided={true} verticalAlign="middle">
          {hasDocuments && <Padding />}
          {hasDocuments &&
            this.props.documents.map((document: ITradeDocument, index: number) => (
              <DocumentListItem
                key={`${document._id} ${index}`}
                document={this.createDocumentItem(document)}
                renderDropdownOptions={() => this.renderDropdownOptions(index, document)}
              />
            ))}
        </List>
      </div>
    )
  }
}
