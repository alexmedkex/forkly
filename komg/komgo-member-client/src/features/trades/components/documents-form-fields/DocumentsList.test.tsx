import * as React from 'react'
import { shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'

import DocumentsList from './DocumentsList'
import { Accordion, List } from 'semantic-ui-react'
import { DocumentType } from '../../../document-management/store/types/document-type'
import { mockProduct } from '../../../document-management/store/products/mock-data'
import { mockCategories } from '../../../document-management/store/categories/mock-data'
import { DocumentListItem } from '../DocumentListItem'
import Text from '../../../../components/text/Text'
import {
  fakeCommercialContractDocumentWithoutExtension,
  fakeCommercialContractDocumentTypes,
  fakeCommercialContractDocumentWithExtension
} from './faker'

describe('DocumentsList component', () => {
  let defaultProps: any

  beforeEach(() => {
    defaultProps = {
      documents: [fakeCommercialContractDocumentWithoutExtension],
      documentTypes: fakeCommercialContractDocumentTypes,
      inForm: true,
      removeDocument: jest.fn(),
      editDocument: jest.fn()
    }
  })

  it('should render DocumentsList component sucessfully', () => {
    const wrapper = shallow(<DocumentsList {...defaultProps} />)

    expect(wrapper.exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<DocumentsList {...defaultProps} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should find the right number of documents', () => {
    const docName2 = 'secondDocument'
    const newDocuments = [
      fakeCommercialContractDocumentWithoutExtension,
      { ...fakeCommercialContractDocumentWithoutExtension, _id: 'document1', name: docName2 }
    ]
    const wrapper = shallow(<DocumentsList {...defaultProps} documents={newDocuments} />)

    const elements = wrapper
      .find(List)
      .shallow()
      .find(DocumentListItem)

    expect(elements.length).toBe(2)

    const getElement = (index: number) => {
      return elements
        .at(index)
        .shallow()
        .find(List.Item)
        .shallow()
        .find(List.Content)
        .at(1)
        .shallow()
        .find(Text)
        .shallow()
    }

    expect(getElement(0).contains(fakeCommercialContractDocumentWithoutExtension.name)).toBeTruthy()
    expect(getElement(1).contains(docName2)).toBeTruthy()
  })

  it('should add file extension', () => {
    const newDocuments = [fakeCommercialContractDocumentWithExtension]

    const elements = shallow(<DocumentsList {...defaultProps} documents={newDocuments} />)
      .find(List)
      .shallow()
      .find(DocumentListItem)

    expect(elements.length).toBe(1)

    const getElement = (index: number) => {
      return elements
        .at(index)
        .shallow()
        .find(List.Item)
        .shallow()
        .find(List.Content)
        .at(1)
        .shallow()
        .find(Text)
        .shallow()
    }

    expect(getElement(0).contains(`${fakeCommercialContractDocumentWithExtension.name}.pdf`)).toBeTruthy()
  })

  it('should not render edit in dropdown options in case editing trade (document.file = null)', () => {
    const documents = [
      {
        ...fakeCommercialContractDocumentWithoutExtension,
        file: null
      }
    ]

    const wrapper = shallow(<DocumentsList {...defaultProps} documents={documents} />)

    const instance = wrapper.instance() as DocumentsList

    const options = instance.renderDropdownOptions(0, documents[0])

    expect(options.length).toBe(1)
    expect(options[0].value).toBe('Delete')
  })
})
