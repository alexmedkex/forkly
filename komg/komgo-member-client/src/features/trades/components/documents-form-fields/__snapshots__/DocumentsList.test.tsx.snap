// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsList component should match snapshot 1`] = `
.c1 {
  padding-right: 0.5em;
}

.c2 {
  display: inline-block;
  font-size: 14px;
  font-weight: bold;
  color: #000000;
  margin: 0;
}

.c3 {
  display: inline;
  left: 50%;
  margin-right: -50%;
  position: absolute;
  text-align: left;
  line-height: 1.9rem !important;
}

.c0 {
  padding-top: 4px;
}

<div>
  <div
    className="ui divided middle aligned list"
    role="list"
  >
    <div
      className="c0"
    />
    <div
      className="item"
      data-test-id="documentName"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="right floated content"
      >
        <div
          aria-expanded={false}
          className="ui inline dropdown c1"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "paddingRight": 0,
            }
          }
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          />
          <i
            aria-hidden="true"
            className="ellipsis horizontal icon"
            onClick={[Function]}
          />
          <div
            className="left menu transition"
          >
            <div
              aria-checked={false}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Delete
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Edit
              </span>
            </div>
          </div>
        </div>
      </div>
      <img
        className="ui inline right spaced image"
        src="/images/file.svg"
      />
      <div
        className="content"
      >
        <span
          className="c2"
        >
          documentName
        </span>
      </div>
      <div
        className="content c3"
      >
        Commercial Contract
      </div>
    </div>
  </div>
</div>
`;
